### PR TITLE
documentation for enable/use SDS in 1.1

### PR DESCRIPTION
**Pull request by quanjielin**
https://github.com/istio/istio/issues/9035

### CODE REVIEW

File: content/docs/tasks/security/auth-sds/index.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+title: Identity provision through SDS
+description: Shows how to enable SDS(secret discovery service) for istio identity provision.
+weight: 70
+keywords: [security,auth-sds]
+---
+
+This task shows how to enable [SDS(secret discovery service)](https://www.envoyproxy.io/docs/envoy/latest/configuration/secret#config-secret-discovery-service) for istio identity provision.
+
+Today in istio workload identity(cert/key) are generated by Citadel and distributed to sidecars
```
[the remark this answers is not in the document]
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: done.

File: content/docs/tasks/security/auth-sds/index.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+through secret-mount files, this approach has below side effects:
+* Traffic interruption:
+  When cert rotation happens, envoy(pilot agent) needs to restart to pick up key/cert update,
```
[the remark this answers is not in the document]
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: renamed.

File: content/docs/tasks/security/auth-sds/index.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    {{< text bash >}}
+    $ cat install/kubernetes/namespace.yaml > istio-auth-sds.yaml
+    $ cat install/kubernetes/helm/istio-init/files/crd-* >> istio-auth-sds.yaml
+    $ helm dep update --skip-refresh install/kubernetes/helm/istio
+    $ helm template install/kubernetes/helm/istio --name istio --namespace istio-system --values install/kubernetes/helm/istio/values-istio-sds-auth.yaml >> istio-auth-sds.yaml
+    $ kubectl create -f istio-auth-sds.yaml
+    {{< /text >}}
+
+## service-to-service mutual TLS using key/certificate provisioned through SDS
```
Comment (wattli) [inline content/docs/tasks/security/auth-sds/index.md]: It should be [S]ervice-to-service

File: content/docs/tasks/security/auth-sds/index.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
+---
+title: Identity provision through SDS
+description: Shows how to enable SDS(secret discovery service) for istio identity provision.
+weight: 70
+keywords: [security,auth-sds]
+---
+
+This task shows how to enable [SDS(secret discovery service)](https://www.envoyproxy.io/docs/envoy/latest/configuration/secret#config-secret-discovery-service) for istio identity provision.
+
+Today the keys and certificates of Istio workloads are generated by Citadel and distributed to sidecars
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: This is the 1.1 doc set, so change this section to be in the past. Something like:

Prior to Istio 1.1, ...

File: content/docs/tasks/security/auth-sds/index.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+through secret-mount files, this approach has below side effects:
+* Traffic interruption:
+  When certificate rotation happens, envoy(pilot agent) needs to restart to pick up key/certificate update,
+  which causes traffic interruption.
+* Potential security issue:
+  Since key/certificate are file-mounted to sidecars, hackers could remote to sidecar container and steal key/certificate content.
+
+To solve the above issues, starting from 1.1 release, Istio supports identity provision through SDS(secret discovery service).
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: These issues are addressed in Istio 1.1 through the support of SDS to provision identities.

File: content/docs/tasks/security/auth-sds/index.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    $ cat install/kubernetes/namespace.yaml > istio-auth-sds.yaml
+    $ cat install/kubernetes/helm/istio-init/files/crd-* >> istio-auth-sds.yaml
+    $ helm dep update --skip-refresh install/kubernetes/helm/istio
+    $ helm template install/kubernetes/helm/istio --name istio --namespace istio-system --values install/kubernetes/helm/istio/values-istio-sds-auth.yaml >> istio-auth-sds.yaml
+    $ kubectl create -f istio-auth-sds.yaml
+    {{< /text >}}
+
+## service-to-service mutual TLS using key/certificate provisioned through SDS
+
+Follow task in [authentication policy](/docs/tasks/security/authn-policy/) to setup test services.
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: Follow the authentication policy task to...

File: content/docs/tasks/security/auth-sds/index.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+{{< text bash >}}
+$ kubectl create ns foo
+$ kubectl apply -f <(istioctl kube-inject -f samples/httpbin/httpbin.yaml) -n foo
+$ kubectl apply -f <(istioctl kube-inject -f samples/sleep/sleep.yaml) -n foo
+$ ubectl create ns bar
+$ kubectl apply -f <(istioctl kube-inject -f samples/httpbin/httpbin.yaml) -n bar
+$ kubectl apply -f <(istioctl kube-inject -f samples/sleep/sleep.yaml) -n bar
+{{< /text >}}
+
+Verify all mutual TLS requests succeed.
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: Change . to :

File: content/docs/tasks/security/auth-sds/index.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+sleep.foo to httpbin.foo: 200
+sleep.foo to httpbin.bar: 200
+sleep.bar to httpbin.foo: 200
+sleep.bar to httpbin.bar: 200
+{{< /text >}}
+
+## Verifying no secret-mount file is generated
+
+As mentioned, instead of using secret-mount file, key/certificate provisioned by SDS flow are stored in workload sidecar's memory,
+which is more secure and avoid traffic interruption.
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: avoid -> avoids

File: content/docs/tasks/security/auth-sds/index.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+Let's verify no secret-mount file is generated by remote accessing to workload sidecar container that we just deployed.
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: Remove "Let's"
remote -> remotely

File: content/docs/tasks/security/auth-sds/index.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+{{< text bash >}}
+$ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..metadata.name}) -c istio-proxy -n foo  -- /bin/bash
+{{< /text >}}
+
+As you can see there is no secret file mounted at `/etc/certs` folder.
+
+## Cleanup
+
+    {{< text bash >}}
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: This block shouldn't be indented.

File: content/docs/tasks/security/auth-sds/index.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+Let's verify no secret-mount file is generated by remote accessing to workload sidecar container that we just deployed.
+
+{{< text bash >}}
+$ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..metadata.name}) -c istio-proxy -n foo  -- /bin/bash
+{{< /text >}}
+
+As you can see there is no secret file mounted at `/etc/certs` folder.
+
+## Cleanup
+
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: You need one sentence to introduce the following text block.

File: content/docs/tasks/security/auth-sds/index.md (hunk 6 above)
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: done.

File: content/docs/tasks/security/auth-sds/index.md (hunk 5 above)
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: updated.

File: content/docs/tasks/security/auth-sds/index.md (hunk 7 above)
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: done

File: content/docs/tasks/security/auth-sds/index.md (hunk 8 above)
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: updated

File: content/docs/tasks/security/auth-sds/index.md (hunk 9 above)
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: updated

File: content/docs/tasks/security/auth-sds/index.md (hunk 10 above)
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: updated

File: content/docs/tasks/security/auth-sds/index.md (hunk 11 above)
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: added

**Comment by quanjielin** [comment]
thanks @geeknoid for the detailed review, could you pls take another look ? thanks !

File: content/docs/concepts/security/index.md [hunk 12]
```diff
@@ -135,10 +135,10 @@ Currently we use different certificate key provisioning mechanisms for each scen
 
 1. The above CSR process repeats periodically for certificate and key rotation.
 
-### Node Agent in Kubernetes (in development)
+### Node Agent in Kubernetes
 
-In the near future, Istio will use node agent in Kubernetes for certificate and key provision, as shown in the figure below.
-Note that the identity provision flow for on-premises machines is the same so we only describe the Kubernetes scenario.
+Istio provides the option of using node agent in Kubernetes for certificate and key provisioning, as shown in the figure below.
+Note that the identity provision flow for on-premises machines will be similar in near future, we only describe the Kubernetes scenario here.
```
Comment (geeknoid) [inline content/docs/concepts/security/index.md]: provision -> provisioning
in near -> in the near

File: content/docs/tasks/security/auth-sds/index.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
+---
+title: Provisioning Identity through SDS
+description: Shows how to enable SDS (secret discovery service) for istio identity provisioning.
+weight: 70
+keywords: [security,auth-sds]
+---
+
+This task shows how to enable [SDS (secret discovery service)](https://www.envoyproxy.io/docs/envoy/latest/configuration/secret#config-secret-discovery-service) for istio identity provisioning.
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: istio -> Istio

File: content/docs/tasks/security/auth-sds/index.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+
+Prior to Istio 1.1, the keys and certificates of Istio workloads are generated by Citadel and distributed to sidecars
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: are -> were

File: content/docs/tasks/security/auth-sds/index.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+through secret-mount files, this approach has below side effects:
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: has below -> has the following

File: content/docs/tasks/security/auth-sds/index.md [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+* Traffic interruption:
+  When certificate rotation happens, envoy(pilot agent) needs to restart to pick up key/certificate update,
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: envoy(pilot agent -> Envoy (Pilot agent

File: content/docs/tasks/security/auth-sds/index.md [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+  which causes traffic interruption.
+* Potential security issue:
+  Since key/certificate are file-mounted to sidecars, hackers could remote to sidecar container and steal key/certificate content.
+
+These issues are addressed in Istio 1.1 through the support of SDS to provision identities.
+
+Workload requests key/certificate from SDS server(node agent, which runs as per-node daemon set) using k8s service account JWT through
+SDS API, node agent generates private key and send CSR request to Citadel, Citadel verifies the JWT and signs the certificate,
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: SDS -> the SDS

File: content/docs/tasks/security/auth-sds/index.md [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+
+{{< text bash >}}
+$ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..metadata.name}) -c istio-proxy -n foo  -- /bin/bash
+{{< /text >}}
+
+As you can see there is no secret file mounted at `/etc/certs` folder.
+
+## Cleanup
+
+Clean up test services and istio control plane:
```
Comment (geeknoid) [inline content/docs/tasks/security/auth-sds/index.md]: istio -> Istio

File: content/docs/concepts/security/index.md (hunk 12 above)
Comment (quanjielin) [reply content/docs/concepts/security/index.md]: updated.

File: content/docs/tasks/security/auth-sds/index.md (hunk 13 above)
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: done.

File: content/docs/tasks/security/auth-sds/index.md (hunk 14 above)
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: updated.

File: content/docs/tasks/security/auth-sds/index.md (hunk 15 above)
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: done.

File: content/docs/tasks/security/auth-sds/index.md (hunk 16 above)
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: done.

File: content/docs/tasks/security/auth-sds/index.md (hunk 17 above)
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: done.

File: content/docs/tasks/security/auth-sds/index.md (hunk 18 above)
Comment (quanjielin) [reply content/docs/tasks/security/auth-sds/index.md]: updated.

**Pull request merged**